lib/spring/generate/springInitializrGenerator.ts

```diff
@@ -92,66 +120,142 @@ function getUrlParameters(params: SpringInitializrProjectCreationParameters): an
     };
 }
 
-interface SpringInitializrProjectCreationParameters extends SpringProjectCreationParameters {
-    projectType: string;
-
-    dependencies?: string;
-
-    language: string;
-
-    bootVersion?: string;
-
-    packaging: string;
+@Parameters()
+export class SpringInitializrProjectCreationParameters implements SmartParameters, SpringProjectCreationParameters {
 
-    javaVersion: string;
-}
-
-const SpringInitializrProjectCreationParameterDefinitions: ParametersObject = {
-    projectType: {
+    @Parameter({
         displayName: "Project Type",
         description: "type of project",
-        validInput: "maven-project or gradle-project",
-        pattern: /^(maven-project|gradle-project)$/,
         required: false,
-        defaultValue: "maven-project",
-    },
+    })
+    public projectType?: string;
```

Comment:
Seems you've lost the default values. You can keep them by making this line:

```typescript
public projectType: string = "maven-project";
```

Similar for the remaining parameters.

Reply:
done

Reply:
We're not using the defaults anymore as these might change in the spring boot initializr. We could use the defaults provided in the metadata, but that's completely redundant.